packages/service-worker/src/services/accounts/AccountsService.ts

```diff
@@ -105,6 +107,7 @@ export class AccountsService implements OnLock, OnUnlock {
     private ledgerService: LedgerService,
     private walletConnectService: WalletConnectService,
     private addressResolver: AddressResolver,
+    private glacierService: GlacierService,
```

Comment:
Instead of the glacier service, we can use the moduleManager, load the EVM module with avalanche C-chain and fetch the history using the module.